### PR TITLE
pass gesture change to shader

### PR DESCRIPTION
**Pull request by bastipnt**

### CODE REVIEW

File: packages/ton-client/src/shaders/render.fs
Comment (felixniemeyer) [inline packages/ton-client/src/shaders/render.fs]: uniforms are global. 
We #include the sdf function. This is similar to C++ simply inserting that file in this shader here. 
So, you don't have to pass uniforms around. 
You can directly access them in the sdf file and function and you should do so in my opinion

File: packages/ton-client/src/shaders/sample.fs
Comment (felixniemeyer) [inline packages/ton-client/src/shaders/sample.fs]: try to move any calculations like these into the CPU thread. 
We don't want to do calculations like `0.75 + 0.15 * x` millions of times in shader invocations when it's sufficient to do them once on the CPU before passing them as uniforms.

**Review by felixniemeyer — CHANGES_REQUESTED** [review]
I made some comments about things that should be discussed before merging.

File: packages/ton-client/src/shaders/render.fs
Comment (felixniemeyer) [reply packages/ton-client/src/shaders/render.fs]: we could also try to define uniforms at the top of the sdf file. 
this may require us to move the include up above the main function. 
this may mess with line numbers on glsl errors (which makes debugging shaders tedious) - I'd like to try whether this is the case

File: packages/ton-client/src/utils/gestures.ts
Comment (felixniemeyer) [inline packages/ton-client/src/utils/gestures.ts]: ooof, I wrote a comment here, too, but I didn't click a button here but directly on review changes in the upper right corner, so this comment was lost -.- 
Let me write it again: 

I think we might want to use `TAP_RADIUS` instead of `TAP_THRESHOLD`: only if a touch moves away at least `TAP_RADIUS` pixels form the initial position it is considered as something else than a tap. 

When using a time based `TAP_THRESHOLD` I can see scenarios in which it fails: 
- e.g. when swiping really fast ('flicking'), e.g. to scroll a large list really quickly, the touch might be shorter then 250 ms and interpreted as a tap. 
- also, when tapping really intently (imagine maybe elders not very familiar with touchscreens), the contact with the screen might be longer than 250 ms and interpreted as a drag or so, while a tap was intended. 

I can implement this, if you want. It's as simple as storing the original position and a boolean `dragging` that get's set to true once the touch moves beyond the `TAP_RADIUS`. If you move back in, dragging remains true, otherwise you'd snap around the initial position when dragging something or similar.

**Review by felixniemeyer — COMMENTED** [review]
added another comment about distinguishing taps from drags

File: packages/ton-client/src/shaders/sdf.glsl
Comment (felixniemeyer) [inline packages/ton-client/src/shaders/sdf.glsl:47]: ooof, I just realized another comment wasn't saved. 

I said here, that you should have a look at line 11 at the function `diamonds` and how I do a scaling there. 

If you want scale to apply to all the sdf, I think you can divide `vec3 p` by `scale` before using it and multiply with `scale` afterwards. 

I know, it's just a test, but if you use it as the second parameter of sdBox in the end, you should call it `boxSize` throughout. If you call it scale, make it behave like scale.

File: packages/ton-client/src/shaders/basti-test.fs
Comment (felixniemeyer) [inline packages/ton-client/src/shaders/basti-test.fs]: if I see it correctly, this shader file is not being used anywhere. 
Do we need it on main then?

Comment (bastipnt) [reply packages/ton-client/src/shaders/basti-test.fs]: added it by accident, was just for me to test, will remove it agin

File: packages/ton-client/src/shaders/sdf.glsl
Comment (bastipnt) [reply packages/ton-client/src/shaders/sdf.glsl:47]: I will have a look tomorrow, thanks

File: packages/ton-client/src/utils/gestures.ts
Comment (bastipnt) [reply packages/ton-client/src/utils/gestures.ts]: changed it, it was also just a first try to make something work

File: packages/ton-client/src/shaders/sample.fs
Comment (bastipnt) [reply packages/ton-client/src/shaders/sample.fs]: yes, I agree, thanks

File: packages/ton-client/src/shaders/render.fs
Comment (bastipnt) [reply packages/ton-client/src/shaders/render.fs]: will try it tomorrow

File: packages/ton-client/src/store.ts
Comment (felixniemeyer) [inline packages/ton-client/src/store.ts:12]: It's probably better to use a class for the state here. 

eg:  
```
class TouchState {
  constructor(
      public tapState: float, 
      public swipeState: float, 
      public pinchState: float
  ) {}
}
```
Or can you explain what a vec3.normalize(state) means? xD

**Review by felixniemeyer — APPROVED** [review]

File: packages/ton-client/src/store.ts
Comment (felixniemeyer) [reply packages/ton-client/src/store.ts:12]: I think I know where you're coming from: 
```
gl.uniform3fv(renderUniLocs.touchManipulationState, store.state);
```
Here you are passing a vec3. 
But you can also pass single floats as uniforms: 
```
gl.uniform1f(renderUniLocs.touchManipulationState, store.tapState);
...
```
Which is nicer because you also get more reasonable variable names in the shader or do not have to deconstruct them: 
`touchManipulationState.x` would become `tapState`

Comment (bastipnt) [reply packages/ton-client/src/store.ts:12]: yeah true, will probably do it like that.

**Pull request merged**
